src/domain/task/usecase/register-task/registerTask.ts

```diff
@@ -1,17 +1,19 @@
 import type { ReduxStore, ThunkResult } from 'domain/task/store/store'
-import type { Task } from 'domain/task/entity/task'
 import { RegisterTaskActions } from './actionCreators'
 import { ThrowErrorActions } from 'domain/common/actionCreators'
 import { ErrorMapper } from 'domain/error/mapper/error.mapper'
 import { UnspecifiedError } from 'domain/task/entity/error'
+import type { CreateTask } from 'domain/task/command/createTask'
+import { TaskMapper } from 'adapters/task/mapper/task.mapper'
+import type { DeepReadonly } from 'superTypes'
 
 const dispatchError = (error: unknown, dispatch: ReduxStore['dispatch']): void => {
   const errorToDispatch = ErrorMapper.mapRawErrorToEntity(error)
   dispatch(ThrowErrorActions.errorThrown(errorToDispatch))
 }
 
 export const registerTask =
-  (task: Task): ThunkResult<Promise<void>> =>
+  (task: DeepReadonly<CreateTask>): ThunkResult<Promise<void>> =>
```

Comment:
If the UC is registerTask, I would expect the command to be named `RegisterTask`.